File: Apputils/aplib/src/main/java/com/allen/aplib/utils/LoggerUtils.java
package com.allen.apputils.utils;

import android.util.Log;

/**
 * Created by Android Studio.
 * User: Allen
 * Date: 2019/4/9
 * Time: 11:37
 */
public class LoggerUtils {
    static boolean DEBUG = true;
    static String TAG = LoggerUtils.class.getClass().getName();

    public static void setDEBUG(boolean DEBUG) {
        LoggerUtils.DEBUG = DEBUG;
    }

    public static void v(String msg){
        if(DEBUG){
            Log.v(TAG,msg);
        }
    }
    public static void d(String msg){
        if(DEBUG){
            Log.d(TAG,msg);
        }
    }
    public static void i(String msg){
        if(DEBUG){
            Log.i(TAG,msg);
        }
    }
    public static void e(String msg){
        if(DEBUG){
            Log.e(TAG,msg);
        }
    }



    public static void v(String tag,String msg){
        if(DEBUG){
            Log.v(tag,msg);
        }
    }
    public static void d(String tag,String msg){
        if(DEBUG){
            Log.d(tag,msg);
        }
    }
    public static void i(String tag,String msg){
        if(DEBUG){
            Log.i(tag,msg);
        }
    }
    public static void e(String tag, String msg){
        if(DEBUG){
            Log.e(tag,msg);
        }
    }
}

File: Apputils/aplib/src/main/java/com/allen/aplib/utils/LoggerUtils.kt
package com.allen.aplib.utils

import android.util.Log

/***
 * create by Allen on 2019/1/2
 */
class LoggerUtils {
    companion object {
        var TAG = "LoggerUtils"
        private var DEBBUG = true
        fun setDebbug(flag: Boolean) {
            DEBBUG = flag
        }

        fun e(title: String) {
            if (DEBBUG) {
                Log.e(TAG, title)
            }
        }

        fun d(title: String) {
            if (DEBBUG) {
                Log.d(TAG, title)
            }
        }

        fun i(title: String) {
            if (DEBBUG) {
                Log.i(TAG, title)
            }
        }

        fun v(title: String) {
            if (DEBBUG) {
                Log.v(TAG, title)
            }
        }


        fun e(tag: String, title: String) {
            if (DEBBUG) {
                Log.e(tag, title)
            }
        }

        fun d(tag: String, title: String) {
            if (DEBBUG) {
                Log.d(tag, title)
            }
        }

        fun i(tag: String, title: String) {
            if (DEBBUG) {
                Log.i(tag, title)
            }
        }

        fun v(tag: String, title: String) {
            if (DEBBUG) {
                Log.v(tag, title)
            }
        }
    }
}